test/extended/syncer/syncer_utils.go

```diff
@@ -0,0 +1,136 @@
+		"--syncer-image=ghcr.io/kcp-dev/kcp/syncer:"+kcpServerVersion, "--output-file="+s.OutputFilePath).Output()
+	if successFlag {
+		o.Expect(createError).ShouldNot(o.HaveOccurred())
+		o.Expect(msg).Should(o.ContainSubstring(containsMsg))
+	} else {
+		o.Expect(createError).Should(o.HaveOccurred())
+		o.Expect(fmt.Sprint(msg)).Should(o.ContainSubstring(containsMsg))
+	}
+}
+
+// GetFieldByJSONPath gets specific field value of the SyncTarget by jsonpath
+func (s *SyncTarget) GetFieldByJSONPath(k *exutil.CLI, JSONPath string) (string, error) {
+	return k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("get").Args("--server="+s.WorkSpaceServer, "syncTarget", s.Name, "-o", "jsonpath="+JSONPath).Output()
+}
+
+// CheckReady checks whether the SyncTarget is ready
+func (s *SyncTarget) CheckReady(k *exutil.CLI) (bool, error) {
+	readyFlag, err := s.GetFieldByJSONPath(k, `{.status.conditions[?(@.type=="Ready")].status}`)
+	if err != nil {
+		e2e.Logf(`Get SyncTarget/%s status faied: "%v"`, s.Name, err)
+		return false, err
+	}
+	e2e.Logf("SyncTarget/%s ready condition is %s", s.Name, readyFlag)
+	return strings.EqualFold(readyFlag, "True"), nil
+}
+
+// WaitUntilReady waits the SyncTarget become ready
+func (s *SyncTarget) WaitUntilReady(k *exutil.CLI) {
+	err := wait.Poll(5*time.Second, 180*time.Second, func() (bool, error) {
+		SyncTargetReady, err := s.CheckReady(k)
+		if err != nil {
+			return SyncTargetReady, err
+		}
+		return SyncTargetReady, nil
+	})
+	exutil.AssertWaitPollNoErr(err, fmt.Sprintf("Waiting SyncTarget/%s become ready timeout", s.Name))
+}
+
+// CheckDisplayAttributes checks the SyncTarget info showing the expected columns
+func (s *SyncTarget) CheckDisplayAttributes(k *exutil.CLI) {
+	syncTargetInfo, err := k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("get").Args("--server="+s.WorkSpaceServer, "synctarget", s.Name).Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(syncTargetInfo).Should(o.And(
+		o.ContainSubstring("NAME"),
+		o.ContainSubstring("AGE"),
+	))
+	syncTargetInfo, err = k.WithoutNamespace().WithoutKubeconf().Run("get").Args("synctarget", s.Name, "-o", "wide").Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(syncTargetInfo).Should(o.And(
+		o.ContainSubstring("NAME"),
+		o.ContainSubstring("LOCATION"),
+		o.ContainSubstring("READY"),
+		o.ContainSubstring("SYNCED"),
+		o.ContainSubstring("API"),
+		o.ContainSubstring("RESOURCES"),
+		o.ContainSubstring("KEY"),
+		o.ContainSubstring("AGE"),
+	))
+}
+
+// Delete the SyncTarget
+func (s *SyncTarget) Delete(k *exutil.CLI) {
+	err := k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("delete").Args("--server="+s.WorkSpaceServer, "synctarget", s.Name).Execute()
+	o.Expect(err).NotTo(o.HaveOccurred())
+}
+
+// Clean the SyncTarget resource
+func (s *SyncTarget) Clean(k *exutil.CLI) {
```

Comment:
what is the diff between Delete & clean here ? Could not really understand.

Reply:
Clean is just used for we don't care the delete result like defer clean, while delete has an assert check that the delete is successful.

Reply:
@Phaow understood that, but my question is mostly about why we need both of them ? Can we just go with delete ?

Reply:
@kasturinarra From my understanding, we need both of them that if we want to check the resources' lifecycle we need the Delete that need check the delete action is successful. In defer logic we just want to recover the env as possible as we can we don't need to make sure the delete action is successful. Clean the ws will also clean its resources. WDYT?

Reply:
Update code make the 2 methods differences more cleary.

Reply:
@Phaow but i think if we do not make sure delete action is successful would not it cause problem for the next test case which is going to run ?

Reply:
Offline discuss addressed. https://coreos.slack.com/archives/C038M7TFEUF/p1662717612446029